src/main/scala/cromwell/binding/WdlBinding.scala

```diff
@@ -115,6 +114,11 @@ object WdlBinding {
     }
   }
 
+  /* All MemberAccess ASTs that are not contained in other MemberAccess ASTs */
+  def findTopLevelMemberAccesses(expr: AstNode) = WdlBinding.findAstsWithTrail(expr, "MemberAccess").filter {
+    case (k, v) => !(v map {case a:Ast => a.getName} contains "MemberAccess")
```

Comment:
What I mean is that outside of our codebase you don't typically see constructs like this:

    x.toSeq.doSomething.filter {
            case z if z > 5 => doBlah
            case x if x > 3 => doFoo
    }

And you don't generally see blocks like this:

    x.map {
          do something
          do something else
          ....
          ....
          ....
          .... (repeat multiple times) 
     }

Yet these sorts of patterns exist all over the place in ours. These constructs have their place but they should be the exception, not the rule. Generally chains of stuff and higher order functions should be pretty simple, ideally on one line. If they start spanning multiple lines it's a sign that perhaps another path should be taken.